tests/test_androidmodule_ctypes_finder.py

```diff
@@ -0,0 +1,158 @@
+
+# Import android._ctypes_library_finder module which we need to do isolated.
+# (since android.__init__ will crash if imported not on android!)
+import shutil
+import tempfile
+temp_d = tempfile.mkdtemp(prefix="p4a-test-ctypes-library-finder-")
+try:
+    import os
+    android_module_folder = os.path.abspath(os.path.join(
+        os.path.dirname(__file__),
+        "..", "pythonforandroid", "recipes", "android", "src"
+    ))
+
+    # Fake android module (with just android._ctypes_library_finder alone):
+    os.mkdir(os.path.join(temp_d, "android"))
+    with open(os.path.join(temp_d, "android", "__init__.py"), "w"):
+        pass
+    shutil.copyfile(os.path.join(android_module_folder,
+                                 "android/_ctypes_library_finder.py"),
+                    os.path.join(temp_d, "android",
+                                 "_ctypes_library_finder.py"))
+
+    # Import the module:
+    import sys
+    sys.path.insert(0, temp_d)
+    import android._ctypes_library_finder   # actual import
+    sys.path.remove(temp_d)
+finally:
+    shutil.rmtree(temp_d)
+
+
+# Fake jnius module with autoclass function:
+import sys
+import types
+module_name = 'jnius'
+bogus_module = types.ModuleType(module_name)
+sys.modules[module_name] = bogus_module
+bogus_module.autoclass = lambda name: None
+
+
+def _get_autoclass_with_fake_activity(returning_test_path="/testpath",
+                                      return_as_service=False):
+    """
+    Helper functions to mock autoclass() from jnius.
+
+    This will return a mocked activity "org.kivy.android.PythonActivity"
+    which supports .getApplicationContext().getPackagemanager(),
+    with that returned object supporting .getApplicationInfo()
+    for PackageManager's GET_SHARED_LIBRARY_FILES info.
+
+    (That is the functionality the android._ctypes_library_finder module
+     uses to scan the application-specific library directories)
+    """
+    try:
+        from unittest.mock import MagicMock
+    except ImportError:  # Python 2.
+        from mock import MagicMock
```

Comment:
FYI `mock` is in are test deps, so even for python3 it gets installed. I know it feels weird since it's already in the core via unittest.mock. But I think it's a common practice still and then you can skip the try/except and go directly for:
```python
from mock import MagicMock
```
`tox` should pass because it `pip install mock`

Reply:
fixed it! thanks for the poke, it does look better that way :+1: